### PR TITLE
Add device daemon

### PR DESCRIPTION
**Pull request by albanD**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131814

Base implementation aiming towards https://github.com/pytorch/rfcs/pull/64

Details of the implementation and next steps in https://github.com/pytorch/pytorch/blob/gh/albanD/3/head/test/cpp_extensions/open_registration_extension/README.md

### CODE REVIEW

File: test/cpp_extensions/open_registration_extension/setup.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def run(self):
         )
     ]
 
+    print(ext_modules)
```
Comment (ezyang) [inline test/cpp_extensions/open_registration_extension/setup.py:47]: print

**Review by ezyang — APPROVED** [review]
Fine. Though I am a bit doubtful how useful this will be as tutorial code though.

**Comment by FFFrog** [comment]
I have another question, this is just my opinion.

The name "PrivateUse1" is not very clear and formal, can it be renamed to something else that makes more sense?

**Comment by albanD** [comment]
Thanks for the comments!

> if we could implement this slef-contained example with a regular way of out-of-tree backend, which would be a bit more complicated, but could simulate a real out-of-tree backend. 

I agree that the standalone repo would be more realistic and easier for users to re-use but I think it will be more complex to integrate into the testing suite as of today. Using a version like this in core will make our life a lot easier for that.

> The name "PrivateUse1" is not very clear and formal, can it be renamed to something else that makes more sense?

I 100% agree. Let me see what we can do here.

**Pull request closed without merging**